libs/dfi/src/dyn_type.c

```diff
@@ -452,64 +389,60 @@ static int dynType_parseSequence(FILE *stream, dyn_type *type) {
     type->sequence.seqType.size = 0;
     type->sequence.seqType.alignment = 0;
 
-    status = dynType_parseWithStream(stream, NULL, type, NULL, &type->sequence.itemType);
-
-    if (status == OK) {
-        type->ffiType = &type->sequence.seqType;
-        dynType_prepCif(&type->sequence.seqType);
+    status = dynType_parseWithStreamOfName(stream, NULL, type, NULL, &type->sequence.itemType, dynType_parseAny);
+    if (status != OK) {
+        return status;
     }
 
-    return status;
+    type->ffiType = &type->sequence.seqType;
+    (void)ffi_get_struct_offsets(FFI_DEFAULT_ABI, type->ffiType, NULL);
```

Comment:
Nice. Previously this was not done.. was this not needed in most case due to an eventual call to `ffi_prep_cif`?

Reply:
Previously, it is done by `dynType_prepCif`, which in turn calls `ffi_prep_cif` and is removed from `dyn_type_common` by this PR.
`ffi_get_struct_offsets` is more explicit about the real intention.

> was this not needed in most case due to an eventual call to ffi_prep_cif?

If used in RSA (i.e. together with `dyn_funciton`), it is indeed not needed as there will be ffi_prep_cif.
If used separately or used for message serialization, it is still necessary.